### PR TITLE
[State transitions] not duplicate

### PR DESCRIPTION
**Pull request by Nil20**

### CODE REVIEW

File: packages/commons/src/fhir/index.ts [hunk 1]
```diff
@@ -401,3 +401,29 @@ export function toHistoryResource<T extends Saved<Resource>>(resource: T) {
...
+              `Composition/${duplicateId}`)
+        )
+      })
+    composition.relatesTo = updatedRelatesTo
+    return composition
+  } else {
+    composition.relatesTo = []
+    return composition
+  }
+}
```
Comment (Zangetsu101) [inline packages/commons/src/fhir/index.ts:429]: Is this function needed in more than one microservice? If not then we can just move it to where it's being used

File: packages/gateway/src/workflow/index.ts [hunk 2]
```diff
@@ -223,6 +224,16 @@ export function certifyRegistration(
   )
 }
 
+export async function markNotAsDuplicate(id: string, authHeader: IAuthHeader) {
```
Comment (Zangetsu101) [inline packages/gateway/src/workflow/index.ts:227]:
```suggestion
export async function markNotADuplicate(id: string, authHeader: IAuthHeader) {
```

File: packages/workflow/src/config/routes.ts [hunk 3]
```diff
@@ -111,6 +112,15 @@ export const getRoutes = () => {
         tags: ['api'],
         description: 'Unassign record endpoint'
       }
+    },
+    {
+      method: 'POST',
+      path: '/record/{id}/not-duplicate',
```
Comment (Zangetsu101) [inline packages/workflow/src/config/routes.ts:118]:
```suggestion
      path: '/records/{id}/not-duplicate',
```
to keep the routes consistent

File: packages/workflow/src/records/search.ts [hunk 4]
```diff
@@ -8,11 +8,19 @@
...
+  Resource,
+  SavedTask,
+  ValidRecord
+} from '@opencrvs/commons/types'
 import { SEARCH_URL } from '@workflow/constants'
 import fetch from 'node-fetch'
 
-export async function indexBundle(bundle: ValidRecord, authToken: string) {
+export async function indexBundle(
+  bundle: ValidRecord | Bundle<Resource>,
```
Comment (Zangetsu101) [inline packages/workflow/src/records/search.ts:21]: Could we keep note of the places that will need to be changed once search stops accepting task bundles?

**Comment by Nil20** [comment]
It'd be better to https://github.com/opencrvs/opencrvs-core/pull/6479 merge this first and then resolve conflicts over here since that ticket has some refactoring of task creation.

File: packages/workflow/src/records/fhir.ts [hunk 5]
```diff
@@ -913,6 +913,27 @@ export function createArchiveTask(
...
+          'http://opencrvs.org/specs/extension/witness-one-signature',
+          'http://opencrvs.org/specs/extension/witness-two-signature'
+        ].includes(extension.url)
+      )
+    ].concat({
+      url: 'http://opencrvs.org/specs/extension/markedAsNotDuplicate'
+    })
+  }
+}
+
```
Comment (Zangetsu101) [inline packages/workflow/src/records/fhir.ts:936]: We should be able to use the common task creation function

File: packages/workflow/src/records/handler/not-duplicate.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  const token = getToken(request)
+  const record = await getRecordById(recordId, token, ['READY_FOR_REVIEW'])
+
+  const notDuplicateBundle = await toNotDuplicated(record, token)
+
+  await indexBundleForAssignment(
+    notDuplicateBundle,
+    token,
+    '/events/not-duplicate'
+  )
```
Comment (Zangetsu101) [inline packages/workflow/src/records/handler/not-duplicate.ts:33]: We should be able to use the usual index bundle function

Comment (Nil20) [reply packages/workflow/src/records/handler/not-duplicate.ts:33]: The assignment property becomes null when we just index the bundle. We don't want that.

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**